src/overlays/actors/ovl_En_Encount1/z_en_encount1.h

```diff
@@ -22,16 +29,16 @@ typedef struct EnEncount1 {
     /* 0x000 */ Actor actor;
     /* 0x144 */ EnEncount1ActionFunc actionFunc;
     /* 0x148 */ Path* path;
-    /* 0x14C */ s16 unk_14C;
-    /* 0x14E */ s16 unk_14E;
-    /* 0x150 */ s16 actorType;
-    /* 0x152 */ s16 unk_152;
-    /* 0x154 */ s16 unk_154;
-    /* 0x156 */ s16 unk_156;
-    /* 0x158 */ s16 unk_158;
-    /* 0x15A */ s16 unk_15A;
-    /* 0x15C */ s32 unk_15C;
-    /* 0x160 */ f32 unk_160;
+    /* 0x14C */ s16 maxSpawnsCur;
+    /* 0x14E */ s16 curSpawnNum;
```

Comment:
Having `cur` as the prefix and suffix here looks a bit weird, but I suppose the `maxSpawns` took priority in terms of naming these right?

Reply:
For myself, I'd need a clearer picture of what the following names represent

* maxSpawnsCur
* curSpawnNum
* totalSpawnNum
* maxSpawnsTotal

Reply:
Yea I don't like a number of variable names here. My current suggestions for renames are these:

* curSpawnNum => numChildren
* maxSpawnsCur => maxChildren
* totalSpawnNum => numSpawned
* maxSpawnsTotal => maxSpawns

Will say that I'm only 90% sure on the names `numSpawned`, `maxSpawns` but I think this brings a lot more clarity on how the values should be used.

Reply:
Yeah `maxSpawns` did take prority, I 
Tbh, I'm not really a fan of just children and think we can be a bit more descriptive. I'm thinking perhaps something more along
-  `curSpawnNum` -> `spawnActive`
- `maxSpawnsCur` -> `spawnActiveMax`
- `totalSpawnNum` -> `spawnTotal`
- `maxSpawnsTotal` -> `spawnTotalMax`

Will probably also adjust the other spawn properties to have a similar format `spawn<Property><Max/Min>`

Reply:
I think just `spawnActive` makes it sound boolean. What about `spawnActiveCount`?

Reply:
Hard disagree with `spawnActive`/`spawnActiveMax`. The whole point of calling it `numChildren`/`maxChildren` is to associate it to the parent/child actor system since child instances access the count in their destructors, and I strongly feel it much more succinctly explains usage and scope in an unambiguous way with a pattern that can be applied to other actors. Trying to force "spawn" into the name only serves to make it harder to distinguish the two counts, and from my perspective it does not add any info that isn't already expressed with the world children.

Reply:
An actor can only have 1 child at a time, so implying `EnCount` can have multiple children, when there's only 1 `/* 0x124 */ struct Actor* child;` would be confusing with `numChildren`

Reply:
You're correct in that the base Actor struct only holds a single child pointer, but I would argue that it would be very confusing to say that a set of child actors with a common parent reference aren't children, or an actor that reserves memory to store multiple child references isn't a parent because it calls `ActorSpawn` over `ActorSpawnAsChild`.

The reality is that actors are allowed to use `Actor.parent` and `Actor.child` as they see fit, and as such you end up with quite a lot of variance with how these values are used. There's nothing stopping you from using the one child pointer to create a circular linked list, i.e. from parent -> child1 -> child2 -> ... -> child1; it's just rather inefficient to do so. As it turns out, it's possible for a circular list of children to occur randomly in OoT's `ObjMure_SetFollowTargets`, the parent instance just never needs to reference the actors via `Actor.child`

Because of this, I would not avoid the terminology.

Reply:
Not sure about the resolution to this, but I think `spawnTotal` -> `spawnTotalCount` to be consistent with `spawnActiveCount`?

Reply:
Fwiw I'm still not a fan of using children. I understand that is the system it uses, but spawn total and spawn active just seem more descriptive in my mind explaining exactly what it is, rather than just a generic child pointer that you than have to go read how are the children used.